Newsletters: add #111 (2020-08-19) (#453)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2020-08-19-newsletter.md

specification of schnorr signatures and a new proposed BIP that
specifies how Bitcoin nodes should handle P2P protocol feature
negotiation in a forwards-compatible way.  Also included are our
regular sections with notable changes to services and client software,
releases and release candidates, and changes to popular Bitcoin
infrastructure software.

## Action items

*None this week.*

## News

- **Proposed uniform tiebreaker in schnorr signatures:** the [elliptic
  curve cryptography][] used in Bitcoin requires identifying points on
  Elliptic Curves (ECs).  This can be done unambiguously using a 32-byte X
  coordinate and a 32-byte Y coordinate.  However, if you know a point's
  X coordinate, then you can calculate the only two possible locations
  for its Y coordinate.  This small ambiguity can be resolved either by
  providing a single bit of extra data to indicate which of the two
  coordinates to use or by using a type of method known as a
  *tiebreaker* to automatically choose one Y coordinate or the other for
  each X coordinate.

    The [BIP340][] specification of schnorr signatures for Bitcoin
    currently uses a tiebreaker method to minimize the amount of data

>     needed to identify EC points in [taproot][topic taproot] public keys

Reviewer: I think just "public keys and signatures" is better than "taproot public keys and schnorr signatures" here.

Author: Why?  (I'm curious, and because my thinking was to make it clear this doesn't change anything for existing SEC-style pubkeys and DER-encoded signatures).

Reviewer: Because BIP340 is separate from taproot, so these changes really apply to all BIP340 pubkeys and signatures. I don't think the word taproot is needed to qualify pubkeys, and since BIP 340 is all about schnorr signatures, I don't think the word schnorr is needed to qualify signatures.

  negotiation]).  The protocol change is simple: when a node receives a
  P2P `version` message from a newly-connected peer, the node should
  ignore any messages it doesn't understand until a `verack` message has
  been received from that peer.  Those messages, like the new BIP339
  `wtxidrelay` message, may signal what features the remote peer
  supports.

    This behavior is already implemented in an unreleased development
    version of Bitcoin Core.  If the maintainers of other
    implementations, or anyone else, opposes this change, please respond
    to the mailing list thread.

## Changes to services and client software

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

FIXME:bitschmidty

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [LND 0.11.0-beta.rc4][lnd 0.11.0-beta] is a release candidate for the
  next major version of this project.  It allows accepting [large
  channels][topic large channels] (by default, this is off) and contains

>   numerous improvements to its backend features that may be of interest

Author: Git grep says I consistently use both "backend" and "frontend".  The only inconsistency in the site is where someone else used "front end".  I don't care much either way, although the versions with the spaces in them look weird to me.  :man_shrugging:

[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #19658][] Allow RPC to fetch all addrman records and add records to addrman FIXME:dongcarl

- [Bitcoin Core #18654][] adds a new `psbtbumpfee` RPC that takes the
  txid of a transaction currently in the local node's mempool and
  creates a [PSBT][topic psbt] that increases its feerate (either by an
  automatic amount or by an amount specified via a parameter).  Other
  RPCs or external tools can then sign and finalize the PSBT.

    The existing `bumpfee` RPC previously created PSBT fee bumps for
    watch-only wallets (see [Newsletter #80][news80 bumpfee]).  This
    behavior is now deprecated and a message will be printed to tell
    those users to use `psbtbumpfee`.  Use of `bumpfee` with regular
    key-containing wallets is unaffected.

- [Bitcoin Core #19070][] causes Bitcoin Core to advertise its support
  for [compact block filters][topic compact block filters] if the

>   `peerblockfilters` configuration option has been enabled (default is

Reviewer 2: both `-peerblockfilters` and `-blockfilterindex` need to be enabled, otherwise bitcoind raises an init error

maybe: "if the `peerblockfilters` and `blockfilterindex` options have been set (default is off)."

  disabled).  This will allow services such as [DNS seeders][] to tell
  lightweight clients the IP addresses of nodes that serve filters.

>   This is the final PR in a series of patches that enable [BIP157][] and

Reviewer 2: As this "This" refers to a different object than the preceding "This", maybe "This PR is the final one"

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #19658][] Allow RPC to fetch all addrman records and add records to addrman FIXME:dongcarl

- [Bitcoin Core #18654][] adds a new `psbtbumpfee` RPC that takes the
  txid of a transaction currently in the local node's mempool and
  creates a [PSBT][topic psbt] that increases its feerate (either by an
  automatic amount or by an amount specified via a parameter).  Other
  RPCs or external tools can then sign and finalize the PSBT.

    The existing `bumpfee` RPC previously created PSBT fee bumps for
    watch-only wallets (see [Newsletter #80][news80 bumpfee]).  This
    behavior is now deprecated and a message will be printed to tell
    those users to use `psbtbumpfee`.  Use of `bumpfee` with regular
    key-containing wallets still creates, signs, and broadcasts the fee bump.

>     This change helps make `bumpfee` more consistent, since its previous

Reviewer 3: ```suggestion
    This change helps make `bumpfee` more consistent since its previous
```

  ignore any messages it doesn't understand until a `verack` message has
  been received from that peer.  Those messages, like the new BIP339
  `wtxidrelay` message, may signal what features the remote peer
  supports.

    This behavior is already implemented in an unreleased development
    version of Bitcoin Core.  If the maintainers of other
    implementations, or anyone else, opposes this change, please respond
    to the mailing list thread.

## Changes to services and client software

*In this monthly feature, we highlight interesting updates to Bitcoin
wallets and services.*

- **Crypto Garage announces P2P Derivatives beta application on Bitcoin:**
  In a [blog post][cg p2p derivatives blog], Crypto Garage outlines a beta
  application for conducting P2P [DLC-based][dlcs] derivatives on Bitcoin regtest
  and testnet. The application allows the specification of a financial
  agreement and creation of a corresponding locking transaction of funds between
  the two parties of that agreement. Upon contract maturity, a price oracle
  provides a signature for the closing transaction spending the amounts
  corresponding to agreement.

- **Specter Desktop adds batching:**

>   [Specter Desktop][specter github], the hardware wallet multisig-focused GUI

Author: ```suggestion
  [Specter Desktop][specter github], a hardware wallet multisig-focused GUI
```

I think Sjors Provoost and others are working on allowing Bitcoin Core's native GUI to access HWI, while Andrew Chow's native descriptor wallet work is making multisig easier, so at some point Bitcoin Core GUI will arguably also be a "hardware wallet multisig-focused GUI".  To prepare for that possible outcome, maybe say "a" rather than "the" here.

  for Bitcoin Core, added the ability to send transactions to multiple receivers.

- **Lightning Labs releases Lightning Terminal:**
  [Lightning Terminal][lightning terminal blog] is a visual, browser-based tool
  for LN channel management, initially focusing on [Lightning Loop][news39 lightning loop announced].

- **Wasabi adds support for PayJoin:**
  [Wasabi 1.1.12][] adds support for the [BIP78][] PayJoin specification that
  also works over Tor.

- **BlueWallet for Desktop alpha announced:**
  BlueWallet announced an [alpha desktop version][bluewallet desktop] for macOS
  of their Lightning and Bitcoin wallet supporting bech32, hardware wallets, PSBTs,

>   watch-only, and more.

Author: ```suggestion
  watch-only addresses, and more.
```

- **BitcoinIsSafe.com lists Bitcoin software marked as malicious by antivirus products:**

>   The [bitcoinissafe.com][] website tracks detection rate of Bitcoin software

Author: ```suggestion
  The [bitcoinissafe.com][] website tracks the detection rate of Bitcoin software
```

  products. The website also provides contact information for notifying
  antivirus vendors about potential false positives.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [LND 0.11.0-beta.rc4][lnd 0.11.0-beta] is a release candidate for the
  next major version of this project.  It allows accepting [large
  channels][topic large channels] (by default, this is off) and contains
  numerous improvements to its backend features that may be of interest
  to advanced users (see the [release notes][lnd 0.11.0-beta]).

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo],
[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #19658][] modifies the `getnodeaddresses` RPC to allow returning

>   all known addresses when 0 is specified as the number of addresses requested.

Author: ```suggestion
  all known addresses when `0` is specified as the number of addresses to retrieve.
```

>   Previously, there was a limit on the maximum number of addresses returned

Author: ```suggestion
  Previously, there was a limit on the maximum number of addresses returned,
```

  mostly due to an internal implementation quirk.

- [Bitcoin Core #18654][] adds a new `psbtbumpfee` RPC that takes the

>   txid of a transaction currently in the local node's mempool and

Reviewer 4: Perhaps helpful to note this is for RBF signaling transactions that are in the wallet as well.
